### PR TITLE
Reorg thing so that base has access to more

### PR DESCRIPTION
**Pull request by duncanspumpkin**
I was having issues with Entity paint setup functions without access to all of the fields this will expose. Originally I was hoping that this sort of arrangement wouldn't be required but oh well! This is how OpenRCT2 handles its entities.

### CODE REVIEW

File: src/OpenLoco/Things/Misc.cpp [hunk 1]
```diff
@@ -35,7 +35,7 @@ exhaust* OpenLoco::exhaust::create(loc16 loc, uint8_t type)
         _exhaust->var_14 = obj->var_05;
         _exhaust->var_09 = obj->var_06;
         _exhaust->var_15 = obj->var_07;
-        _exhaust->type = misc_thing_type::exhaust;
+        _exhaust->mType = MiscThingType::exhaust;
```
Comment (marijnvdwerf) [inline src/OpenLoco/Things/Misc.cpp:38]: Not a fan of the abbreviations. Would rather see `vehicleType` and `miscType`. Or a `getSubType()` and `setSubType()` method, with the `subType` being treated as a 'private' `uint8`. I think the latter has my preference.

Comment (duncanspumpkin) [reply src/OpenLoco/Things/Misc.cpp]: okay what about my new version?

**Comment by marijnvdwerf** [comment]
Is there a (not overcomplicated) way of making both MiscBaseThing and VehicleBaseThing just have a `getSubType()` method, just with a different type? I feel like 'getVehicleType()' gets a bit confusing, since that sounds more like 'airplane' or 'train', rather than components like bogeys and heads.

**Comment by duncanspumpkin** [comment]
I did originally do that but thought it might get confusing if you accidentally called the base instead of the derived

**Comment by duncanspumpkin** [comment]
see latest commit

**Review by marijnvdwerf — APPROVED** [review]
LGTM. Are you okay with it as well?

**Pull request merged**